### PR TITLE
DEV-3954-check-user

### PR DESCRIPTION
**Pull request by mikysett**
add a new exercise to scripting piscine.

Here also the unofficial script mode is turned off in tests because some case scenarios do expect non-zero return statuses

### CODE REVIEW

File: subjects/devops/check-user/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+> Your results may appear slightly different.
+
+### Error handling
+
+All errors will print a specific message on **stderr** (ending with a newline) and returns a specific non-zero value:
+- Wrong number of arguments: `"Error: expect 2 arguments"`, returns `1`.
+- First argument different from `-e` or `-i`: `"Error: unknown flag"`, exit with `1`.
+
+### Hints
```
Comment (MSilva95) [inline subjects/devops/check-user/README.md:36]: You need to give examples of the getent usage and explain a little what is that command

File: subjects/devops/check-user/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+$ ./check-user.sh -i unknown
+$
+```
+
+> Your results may appear slightly different.
+
+### Error handling
+
+All errors will print a specific message on **stderr** (ending with a newline) and returns a specific non-zero value:
+- Wrong number of arguments: `"Error: expect 2 arguments"`, returns `1`.
```
Comment (MSilva95) [inline subjects/devops/check-user/README.md:33]: Why is one written  "returns `1`" and the other is " exit with `1`."?
Can they be written the same way in order to avoid confusion?

File: sh/tests/check-user_test.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+'
+
+script_dirS=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)
+
+challenge() {
+    submitted="./check-user.sh $@
+    "
+    expected="./check-user.sh $@
+    "
+    submitted+=$(2>&1 bash "$script_dirS"/student/check-user.sh "$@")
```
Comment (MSilva95) [inline sh/tests/check-user_test.sh:14]: not sure why but as soon as i format this changes to 
```
submitted+=$(bash 2>&1 "$script_dirS"/student/check-user.sh "$@")
expected+=$(bash 2>&1 "$script_dirS"/solutions/check-user.sh "$@")
```
But it works the same way so no worries here

File: sh/tests/solutions/check-user.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```
Comment (MSilva95) [inline sh/tests/solutions/check-user.sh:1]: normally we are making the solution an executable too, even if it is not necessary since we use "bash", but since it was like this for the Go piscine we also kept this process.

**Review by MSilva95 — CHANGES_REQUESTED** [review]
The test and solution are working well!
Just needs some more guidance for the students

File: subjects/devops/check-user/README.md (hunk 2 above)
Comment (mikysett) [reply subjects/devops/check-user/README.md:33]: done

File: subjects/devops/check-user/README.md (hunk 1 above)
Comment (mikysett) [reply subjects/devops/check-user/README.md:36]: done

File: subjects/devops/check-user/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+- First argument different from `-e` or `-i`: `"Error: unknown flag"`, exit with `1`.
+
+### Hints
+
+> `man getent` will be a great resource to explore
```
[the remark this answers is not in the document]
Comment (mikysett) [reply subjects/devops/check-user/README.md:38]: I've added more hints about `getent`, also the link in the resources have a dedicated session about it, explaining quite well how to use it.

File: sh/tests/check-user_test.sh (hunk 3 above)
Comment (mikysett) [reply sh/tests/check-user_test.sh:14]: I've applied the autoformat, but yes it seems like a convention, not changing much

**Review by MSilva95 — APPROVED** [review]
LGTM
